### ISSUE TITLE
UUID naming

### ISSUE THREAD

**Issue by danielytics** [TheCherno/Hazel#488]
I highly recommend not calling them UUID's, but instead consider UID or UniqueID or similar. The "Universally" (or "Globally" for GUID) has a special meaning. [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) is a well-defined and common term in software and Hazel's "UUID"'s are not these, instead its co-opting the term for a similar but distinct concept. That seems like a cause of confusion.

I suggest simply "UID", since that's what they are: Unique ID's.

**Comment by cugone** [comment]
Additionally a recent update to windows that makes some changes to the Windows SDK causes issues.

I had defined my UUID class as unqualified (not inside a namespace); turns out the new update adds a class to the Windows SDK named `UUID` that is also unqualified.

Needless to say: Things broke.

**Comment by Pikachuxxxx** [comment]
Yep godot calls them RID(resource ID) or even better would be to make UUID an abstract class and use winapi or platform specific api to easily generate 128-bit uuids?

**Comment by neo-mashiro** [comment]
These IDs in theory can also have a collision (although the probability is close to 0), so the name "unique" might be inappropriate as well...
